bot.py

```diff
@@ -82,47 +136,92 @@ def start_game_msg(name):
     GAME_NUM += 1
 
 
-def end_game():
-    while not check_image("rc_items/gain_power.png"):
-        time.sleep(1)
-    click_image("rc_items/gain_power.png")
+def end_game(self, fail=False):
+    if not fail:
+        self.game_status = "idle"
+
+        keyboard.press_and_release("page up")
+        keyboard.press_and_release("down")
+
+        while not check_image("rc_items/gain_power.png"):
+            time.sleep(1)
+        click_image("rc_items/gain_power.png")
+
+        time.sleep(3)
+
+        keyboard.press_and_release("page up")
+        time.sleep(2)
+        click_image("rc_items/goto_games.png")
+        time.sleep(2)
 
-    keyboard.press_and_release("page up")
-    time.sleep(0.5)
-    click_image("rc_items/goto_games.png")
-    time.sleep(0.5)
-    if check_image("rc_items/collect_pc.png"):
-        click_image("rc_items/click_image")
+        if check_image("rc_items/collect_pc.png"):
+            click_image("rc_items/click_image")
+    else:
+        keyboard.press_and_release("page up")
+        time.sleep(2)
+
+        click_image("rc_items/goto_games.png")
+
+        os.execv(sys.executable, ['python'] + sys.argv) #restart script
+
+
+class ThreadWithReturnValue(Thread):
+    def __init__(self, group=None, target=None, name=None,
+                 args=(), kwargs={}, Verbose=None):
+        Thread.__init__(self, group, target, name, args, kwargs)
+        self._return = None
+
+    def run(self):
+        if self._target is not None:
+            self._return = self._target(*self._args,
+                                        **self._kwargs)
+
+    def join(self, *args):
+        Thread.join(self, *args)
+        return self._return
```

Comment:
method for thread with return value (return value will be the matches of CoinFlip)